File: src/main/java/com/example/venka/javaspringsparkdemo/mongo/DatabaseManagerImpl.java
package com.example.venka.javaspringsparkdemo.mongo;

import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public final class DatabaseManagerImpl implements DatabaseManager {

    private final MongoCollection<Document> COLLECTION;

    private DatabaseManagerImpl(@Autowired MongoDatabase mongoDatabase) {
        COLLECTION = mongoDatabase.getCollection("messages");
    }

    @Override
    public void insert(final Document document) {
        COLLECTION.insertOne(document);
    }

    @Override
    public void delete(final Document document) {
        COLLECTION.deleteOne(document);
    }

    @Override
    public List<Document> getAll() {
        return COLLECTION.find().into(new ArrayList<>());
    }

    @Override
    public Document getById(final Document document) {
        return COLLECTION.find(document).first();
    }
}

File: src/main/kotlin/com/example/venka/kotlinspringsparkdemo/mongo/DatabaseManagerImpl.kt
package com.example.venka.kotlinspringsparkdemo.mongo

import com.mongodb.client.MongoCollection
import com.mongodb.client.MongoDatabase
import org.bson.Document
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
class DatabaseManagerImpl(@Autowired private val mongoDatabase: MongoDatabase) : DatabaseManager {
    private final val collection: MongoCollection<Document> = mongoDatabase.getCollection("messages")

    override val all: List<Document>
        get() = collection.find().into(ArrayList())

    override fun insert(document: Document) {
        collection.insertOne(document)
    }

    override fun delete(document: Document) {
        collection.deleteOne(document)
    }

    override fun getById(document: Document): Document {
        return collection.find(document).first()
    }

}